packages/desktop-client/src/components/schedules/DiscoverSchedules.tsx

```diff
@@ -110,33 +123,50 @@ function DiscoverSchedulesTable({ schedules, loading }) {
   );
 }
 
-export default function DiscoverSchedules({ modalProps, actions }) {
-  let { data: schedules, isLoading } =
-    useSendPlatformRequest('schedule/discover');
-  if (!schedules) schedules = [];
-
-  let [creating, setCreating] = useState(false);
-
-  let selectedInst = useSelected('discover-schedules', schedules, []);
+export default function DiscoverSchedules({
+  modalProps,
+  actions,
+}: {
+  modalProps: CommonModalProps;
+  actions: BoundActions;
+}) {
+  const { data, isLoading } = useSendPlatformRequest('schedule/discover') as {
+    data: Awaited<ReturnType<SchedulesHandlers['schedule/discover']>> | null;
+    isLoading: boolean;
+  };
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏we should be able to automatically infer the return type here. If we're not able to: there must be some issue that needs fixing.
```suggestion
  const { data, isLoading } = useSendPlatformRequest('schedule/discover');
```